src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -3659,8 +3659,387 @@ class StubGenerator: public StubCodeGenerator {
+      };
+      const int const_add = vset_sew == Assembler::e32 ? 16 : 32;
+
+      __ align(CodeEntryAlignment);
+      StubCodeMark mark(_cgen, "StubRoutines", stub_name(vset_sew, multi_block));
+      address start = __ pc();
+
+      Register buf   = c_rarg0;
+      Register state = c_rarg1;
+      Register ofs   = c_rarg2;
+      Register limit = c_rarg3;
+      Register consts =  t2; // caller saved
+      Register state_c = x28; // caller saved
+      VectorRegister vindex = v1;
+
+      Label multi_block_loop;
+
+      __ enter();
+
+      address constant_table = vset_sew == Assembler::e32 ? (address)round_consts_256 : (address)round_consts_512;
+      la(consts, ExternalAddress(constant_table));
+
+      // Register use in this function:
+      //
+      // VECTORS
+      //  v10 - v13 (512/1024-bits / 4*128/256 bits / 4*4*32/65 bits), hold the message
+      //             schedule words (Wt). They start with the message block
+      //             content (W0 to W15), then further words in the message
+      //             schedule generated via vsha2ms from previous Wt.
+      //   Initially:
+      //     v10 = W[  3:0] = { W3,  W2,  W1,  W0}
+      //     v11 = W[  7:4] = { W7,  W6,  W5,  W4}
+      //     v12 = W[ 11:8] = {W11, W10,  W9,  W8}
+      //     v13 = W[15:12] = {W15, W14, W13, W12}
+      //
+      //  v16 - v17 hold the working state variables (a, b, ..., h)
+      //    v16 = {f[t],e[t],b[t],a[t]}
+      //    v17 = {h[t],g[t],d[t],c[t]}
+      //   Initially:
+      //    v16 = {H5i-1, H4i-1, H1i-1 , H0i-1}
+      //    v17 = {H7i-i, H6i-1, H3i-1 , H2i-1}
+      //
+      //  v0 = masks for vrgather/vmerge. Single value during the 16 rounds.
+      //
+      //  v14 = temporary, Wt+Kt
+      //  v15 = temporary, Kt
+      //
+      //  v18/v19 = temporaries, in the epilogue, to re-arrange
+      //            and byte-swap v16/v17
+      //
+      //  v26/v27 = hold the initial values of the hash, byte-swapped.
+      //
+      //  v30/v31 = used to generate masks, vrgather indices.
+      //
+      // During most of the function the vector state is configured so that each
+      // vector is interpreted as containing four 32/64 bits (e32/e64) elements (128/256 bits).
+
+      // Set vectors as 4 * 32/64 bits
+      //
+      // e32/e64: vector of 32b/64b/4B/8B elements
+      // m1: LMUL=1
+      // ta: tail agnostic (don't care about those lanes)
+      // ma: mask agnostic (don't care about those lanes)
+      // x0 is not written, we known the number of vector elements.
+
+      __ vsetivli(x0, 4, vset_sew, Assembler::m1, Assembler::ma, Assembler::ta);
+      // Splat indexes in vindex if SEW = e64, but don't hurt anything.
+      int64_t indexes = vset_sew == Assembler::e32 ? 0x00041014ul : 0x00082028ul;
+      __ li(t0, indexes);
+      __ vmv_s_x(vindex, t0);
+
+      // Step-over a,b, so we are pointing to c.
+      // const_add is equal to 4x state variable, div by 2 is thus 2, a,b
```

Comment:
I don't quite understand this code comment.

Reply:
State is: `{a,b,c,d,e,f,g,h}`
The index used assumes the pointer is at 'c'.
As we know const_add is equal to the size of 4 state variables, if we divide it by 2 we have the size of two state variables.

If that make sense, can you suggest a comment change?